packages/server/src/api/routes/tests/row.spec.ts

```diff
@@ -3227,35 +3233,106 @@ datasourceDescribe(
           name: generator.word(),
           description: generator.paragraph(),
         })
-        await config.api.row.save(table._id!, {
+        mainRow = await config.api.row.save(table._id!, {
           name: generator.word(),
           description: generator.paragraph(),
           tableId: table._id!,
           links: [relatedRow._id],
         })
       })
 
+      async function updateFormulaColumn(
+        formula: string,
+        opts?: { responseType?: FormulaResponseType; formulaType?: FormulaType }
+      ) {
+        table = await config.api.table.save({
+          ...table,
+          schema: {
+            ...table.schema,
+            formula: {
+              name: "formula",
+              type: FieldType.FORMULA,
+              formula: formula,
+              responseType: opts?.responseType,
+              formulaType: opts?.formulaType || FormulaType.DYNAMIC,
+            },
+          },
+        })
+      }
+
       it("should be able to search for rows containing formulas", async () => {
         const { rows } = await config.api.row.search(table._id!)
         expect(rows.length).toBe(1)
         expect(rows[0].links.length).toBe(1)
         const row = rows[0]
         expect(row.formula).toBe(relatedRow.name)
       })
+
+      it("should coerce - number response type", async () => {
+        await updateFormulaColumn(encodeJS("return 1"), {
+          responseType: FieldType.NUMBER,
+        })
+        const { rows } = await config.api.row.search(table._id!)
+        expect(rows[0].formula).toBe(1)
+      })
+
+      it("should coerce - boolean response type", async () => {
+        await updateFormulaColumn(encodeJS("return true"), {
+          responseType: FieldType.BOOLEAN,
+        })
+        const { rows } = await config.api.row.search(table._id!)
+        expect(rows[0].formula).toBe(true)
+      })
+
+      it("should coerce - datetime response type", async () => {
+        await updateFormulaColumn(encodeJS("return new Date()"), {
+          responseType: FieldType.DATETIME,
+        })
+        const { rows } = await config.api.row.search(table._id!)
+        expect(isDate(rows[0].formula)).toBe(true)
+      })
+
+      it("should coerce - datetime with invalid value", async () => {
+        await updateFormulaColumn(encodeJS("return 'a'"), {
+          responseType: FieldType.DATETIME,
+        })
+        const { rows } = await config.api.row.search(table._id!)
+        expect(rows[0].formula).toBeUndefined()
+      })
+
+      it("should coerce handlebars", async () => {
+        await updateFormulaColumn("{{ add 1 1 }}", {
+          responseType: FieldType.NUMBER,
+        })
+        const { rows } = await config.api.row.search(table._id!)
+        expect(rows[0].formula).toBe(2)
+      })
```

Comment:
It's a bit nitty but could you add a test case that does `{{ add 1 1 }}` but returns a string? Just to make sure the string get passed through correctly, didn't see that in other tests.